src/cosmos-keys.ts

```diff
@@ -72,9 +72,14 @@ function deriveMasterKey(mnemonic: string): bip32.BIP32 {
   return masterKey
 }
 
-function deriveKeypair(masterKey: bip32.BIP32): KeyPair {
+function deriveKeypair(masterKey: bip32.BIP32Interface): KeyPair {
   const cosmosHD = masterKey.derivePath(hdPathAtom)
   const privateKey = cosmosHD.privateKey
+
+  if (!privateKey) {
```

Comment:
Is there any case were this would happen? Does `derivePath` return `{}` in any case.

Reply:
No. `privateKey` has typing `Buffer | undefined`, so we have to check if it really exists. But I think we can safely cast it to `Buffer` as `privateKey` is always created when using `fromSeed`

Reply:
The latter sounds like a good idea. The current implementation suggests it is undefined in some case.

Reply:
Updated, can you take a look when you will have time, ty